components/SignInOrJoinFree.js

```diff
@@ -135,6 +150,7 @@ class SignInOrJoinFree extends React.Component {
     const { submitting, error, unknownEmailError, email } = this.state;
     const displayedForm = this.props.form || this.state.form;
     const routes = this.props.routes || {};
+    const { intl, enforceTwoFactorAuthForLoggedInUser } = this.props;
 
     return (
       <Flex flexDirection="column" width={1} alignItems="center">
```

Comment:
This `return` block starts to be really huge. Would be interesting to split it in more functions for readability.